### ISSUE TITLE
Double POST with validate/__intent__ ?

### ISSUE THREAD

**Issue by meza** [edmundhung/conform#133]
Using Remix, I'm finding a weird behaviour for a simple newsletter form.

When I submit, I get this:

![image](https://user-images.githubusercontent.com/211977/232502338-69c95a1b-8097-405a-8f01-b864967c0478.png)

The first call behaves as expected
![image](https://user-images.githubusercontent.com/211977/232502451-17682fa5-9263-4f86-94c3-f48009fd0811.png)

It does a redirect('/');

The next invocation however looks like this:

![image](https://user-images.githubusercontent.com/211977/232502585-6cee1da3-386f-4296-bd3a-bea3b4caa429.png)

This obviously fails on the server that's expecting a different intent, thus renders the form invalid - returning the data, filling out the form.

What am I doing wrong?
Where does the validate/__intent__ come from?

My form config:

```typescript

const [form, { email }] = useForm({
    constraint: getFieldsetConstraint(schema),
    fallbackNative: true,
    lastSubmission: lastSubmission,
    shouldValidate: 'onBlur',
    onValidate: function ({ formData }) {
      const result = parse(formData, { schema: schema });
      setIsValid(result.error === undefined);
      return result;
    }
  });

```

Some of the other bits:

```typescript

const NEWSLETTER_INTENT = 'newsletter.signup';

export const schema = z.object({
  email: z.string().min(1, 'validation.email.required').email('validation.email.invalid')
});

export const newsletterSignUpAction: ActionFunction = async ({ request }) => {
  const formData = await request.formData();
  const submission = parse(formData, { schema: schema });

  console.log({submission: submission});

  if (!submission.value || submission.intent !== NEWSLETTER_INTENT) {
    return json(submission, { status: 400 });
  }

  throw redirect('/');
  // do something with result.data
};

```

**Comment by meza** [comment]
It's a bit difficult to show because it's all out in different components + in a private repo but here's the button:

```typescript
import { useEffect, useState } from 'react';
import { conform } from '@conform-to/react';
import classNames from 'classnames';

export interface SubmitButtonProps {
  intent: string;
  className?: string;
  disabled?: boolean;
}

export const SubmitButton = (props: SubmitButtonProps) => {
  const [disabled, setDisabled] = useState(props.disabled || false);

  useEffect(() => {
    setDisabled(props.disabled || false);
  }, [props.disabled]);

  return (
    <button className={classNames('primary', props.className)} type='submit' name={conform.INTENT} value={props.intent} disabled={disabled}>Submit</button>
  );
};

```

The input as follows:

```typescript
export interface InputProps {
  label?: string;
  placeholder: string;
  conformField: FieldConfig;
  tabIndex?: number | undefined;
  type?: HTMLInputTypeAttribute | undefined;
  value?: string | undefined;
}
export const Input = (props: InputProps) => {
  const finalType = props.type || 'text';
  const id = useId();
  const errorId = useId();
  const { t } = useTranslation();

  return (
    <div className={classNames('input')}>
      <input
        className={classNames({ 'error': props.conformField.error !== undefined })}
        {...conform.input(props.conformField, { type: finalType })}
        id={id}
        placeholder={' '}
        tabIndex={props.tabIndex}
        aria-invalid={props.conformField.error === undefined ? 'false' : 'true'}
        aria-required={props.conformField.required === undefined ? false : props.conformField.required}
        aria-describedby={errorId}
      />
      {props.label && <label htmlFor={id} className={classNames({ 'required': props.conformField.required })}>{props.label}</label>}
      <div id={errorId} role='alert' className={classNames('input-error')} aria-live={'polite'}>{props.conformField.error ? t(props.conformField.error) : ''}</div>
    </div>
  );
};
```

And the form is:

```typescript
<newsletter.Form
        className={classNames('newsletter', { 'visible': true })}
        method={'POST'} {...form.props}>
        <Input label={'What is your email address?'} type={'email'} tabIndex={3} conformField={email}/>
        <div className={'button'}>
          <SubmitButton disabled={!!email.error || (newsletter.state === 'submitting')} intent={NEWSLETTER_INTENT}/>
        </div>
</newsletter.Form>
```

**Comment by edmundhung** [comment]
It looks like there is a bug on the revalidation logic currently which validates your submit, very likely when you unfocus the submit button (i.e. onBlur). But this still doesn't answer why you are see the request made there with a client validation in place. The only exception is if it throws an error during validation and so Conform fallbacks to server validation. Can you verify this?

**Comment by meza** [comment]
@edmundhung about to test! Been tied up with other things but let's experiment!

**Comment by meza** [comment]
It would appear that it's all coming from the onBlur. When I change it to onInput, it doesn't do the double post.
It however doesn't clear out the submitted field but that might just be user error :D

**Comment by edmundhung** [comment]
This should be fixed on [v0.6.2](https://github.com/edmundhung/conform/releases/tag/v0.6.2).
